gratipay/models/package/__init__.py

```diff
@@ -40,3 +51,35 @@ def from_names(cls, package_manager, name):
         """
         return cls.db.one("SELECT packages.*::packages FROM packages "
                           "WHERE package_manager=%s and name=%s", (package_manager, name))
+
+
+    @property
+    def team(self):
+        """A computed attribute, the :py:class:`~gratipay.models.team.Team`
+        linked to this package if there is one, otherwise ``None``. Makes a
+        database call.
+        """
+        return self._load_team(self.db)
+
+
+    def ensure_team(self, cursor, owner):
+        """Given a db cursor and :py:class:`Participant`, insert into ``teams`` if need be.
+        """
+        team = self._load_team(cursor)
```

Comment:
Here we go: team is not `None` the second time [around](https://github.com/gratipay/gratipay.com/blob/3d8fb4413b4370181a040f4a3a8f2c094a3bb088/gratipay/models/package/__init__.py#L68-L69).

```
tests/py/test_email.py pid-36042 thread-4463992832 (Thread-2) have cursor, will travel
pid-36042 thread-4463992832 (Thread-2) have record:Record(id=118, address=u'alice@example.com', verified=
None, nonce=u'fccf6dec-7559-4cb2-ba77-046f2aa0efe7', verification_start=datetime.datetime(2017, 4, 24, 13, 44, 6, 588594, tzinfo=psycopg2.tz.FixedOffsetTimezone(offset=0, name=None)), verification_end=None, participant_id=1L)
pid-36042 thread-4463992832 (Thread-2) have packages:[<gratipay.models.package.Package object at 0x109d09250>]
pid-36042 thread-4463992832 (Thread-2) let's finish up ...
pid-36042 thread-4468199424 (Thread-3) have cursor, will travel
pid-36042 thread-4463992832 (Thread-2) finishing up ...
pid-36042 thread-4463992832 (Thread-2) team: None
pid-36042 thread-4468199424 (Thread-3) have record:Record(id=118, address=u'alice@example.com', verified=None, nonce=u'fccf6dec-7559-4cb2-ba77-046f2aa0efe7', verification_start=datetime.datetime(2017, 4, 24, 13, 44, 6, 588594, tzinfo=psycopg2.tz.FixedOffsetTimezone(offset=0, name=None)), verification_end=None, participant_id=1L)
pid-36042 thread-4468199424 (Thread-3) have packages:[<gratipay.models.package.Package object at 0x109d12250>]
pid-36042 thread-4463992832 (Thread-2) inserting into teams_to_packages: 45 95
pid-36042 thread-4468199424 (Thread-3) let's finish up ...
pid-36042 thread-4463992832 (Thread-2) inserted into teams
pid-36042 thread-4463992832 (Thread-2) [Record(team_id=45L, package_id=95L)]
pid-36042 thread-4468199424 (Thread-3) finishing up ...
pid-36042 thread-4468199424 (Thread-3) team: 80e34bf6-324a-4b63-a700-c3fe3b624001
```

```diff
diff --git a/gratipay/models/package/__init__.py b/gratipay/models/package/__init__.py
index a0591a1..9a69037 100644
--- a/gratipay/models/package/__init__.py
+++ b/gratipay/models/package/__init__.py
@@ -65,7 +65,10 @@ class Package(Model):
     def ensure_team(self, cursor, owner):
         """Given a db cursor and :py:class:`Participant`, insert into ``teams`` if need be.
         """
+        from aspen import log
+        log('finishing up ...')
         team = self._load_team(cursor)
+        log('team: {}'.format(team.name if team else 'None'))
         if not team:
             slug = str(uuid.uuid4()).lower()
             team = Team.insert( slug=slug
@@ -76,6 +79,7 @@ class Package(Model):
                               , owner=owner
                               , _cursor=cursor
                                )
+            log('inserting into teams_to_packages: {} {}'.format(team.id, self.id))
             cursor.run('INSERT INTO teams_to_packages (team_id, package_id) '
                        'VALUES (%s, %s)', (team.id, self.id))
             from aspen import log
diff --git a/gratipay/models/participant/email.py b/gratipay/models/participant/email.py
index e47cafe..5ae2528 100644
--- a/gratipay/models/participant/email.py
+++ b/gratipay/models/participant/email.py
@@ -228,13 +228,17 @@ class Email(object):
 
 
     def finish_email_verification(self, email, nonce):
+        from aspen import log
         if '' in (email.strip(), nonce.strip()):
             return VERIFICATION_MISSING
         with self.db.get_cursor() as cursor:
+            log('have cursor, will travel')
             record = self.get_email(email, cursor)
+            log('have record:' + str(record))
             if record is None:
                 return VERIFICATION_FAILED
             packages = self.get_packages_claiming(cursor, nonce)
+            log('have packages:' + str(packages))
             if record.verified and not packages:
                 assert record.nonce is None  # and therefore, order of conditions matters
                 return VERIFICATION_REDUNDANT
@@ -243,6 +247,7 @@ class Email(object):
             if (utcnow() - record.verification_start) > EMAIL_HASH_TIMEOUT:
                 return VERIFICATION_EXPIRED
             try:
+                log('let\'s finish up ...')
                 self.finish_package_claims(cursor, nonce, *packages)
                 self.save_email_address(cursor, email)
                 cursor.connection.commit()

```